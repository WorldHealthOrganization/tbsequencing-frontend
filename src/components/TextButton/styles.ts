import { css } from '@emotion/react';
import appColors from '../../styles/colors';

export const label = css({
  color: appColors.secondary.whoBlue,
  textDecoration: 'underline',
  letterSpacing: '1.25px',
  lineHeight: '24px',
});

export const button = css({
  border: 'none',
  padding: 0,
  background: 'none',
  cursor: 'pointer',
});

export const disabled = css({
  color: appColors.neutral.grey,
  cursor: 'default',
});
