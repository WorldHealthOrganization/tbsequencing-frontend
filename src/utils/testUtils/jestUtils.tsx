import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { reducer } from '../../app/store';

const getWrapper = (
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
  } = {},
) => ({ children }: { children: ReactElement }) => (
  <Provider store={store}>{children}</Provider>
);

function render(
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {},
) {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const getTreeJSONWithRedux = (ui: ReactElement, {
  // @ts-ignore
  preloadedState,
  store = configureStore({ reducer, preloadedState }),
} = {}) => {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  const tree = renderer
    .create(<Wrapper>{ui}</Wrapper>)
    .toJSON();

  return tree;
};

export const wrappedInRouter = (
  component: ReactElement,
) => <BrowserRouter>{component}</BrowserRouter>;

export * from '@testing-library/react';
export { render, getTreeJSONWithRedux };
