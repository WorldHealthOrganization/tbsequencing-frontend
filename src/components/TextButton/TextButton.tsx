/** @jsxImportSource @emotion/react */
import React, { ReactNode, MouseEvent } from 'react';
import { SerializedStyles } from '@emotion/react';
import * as styles from './styles';
import PrimaryText from '../typography/PrimaryText';

interface Props {
  children: ReactNode;
  onClick: (event: MouseEvent<HTMLButtonElement>) => void;
  style?: SerializedStyles;
  disabled?: boolean;
}

export const TextButton = ({
  children, onClick, style, disabled = false,
}: Props) => {
  const disabledStyle = disabled ? styles.disabled : undefined;

  return (
    <button
      disabled={disabled}
      css={[styles.button, style]}
      type="button"
      onClick={onClick}
    >
      <PrimaryText style={[styles.label, disabledStyle]}>{children}</PrimaryText>
    </button>
  );
};

export default TextButton;
