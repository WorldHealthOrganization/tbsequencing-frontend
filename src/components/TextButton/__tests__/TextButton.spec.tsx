import React from 'react';
import { getTreeJSONWithRedux, render, screen } from '../../../utils/testUtils/jestUtils';
import TextButton from '../TextButton';

describe('TextButton component test', () => {
  const mockOnClick = jest.fn();
  const mockLabel = 'mockLabel';

  test('renders TextButton component without error', () => {
    render(
      <TextButton onClick={mockOnClick}>{mockLabel}</TextButton>,
    );
  });

  it('Should render label properly', () => {
    render(
      <TextButton onClick={mockOnClick}>{mockLabel}</TextButton>,
    );

    screen.getByText(mockLabel);
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <TextButton onClick={mockOnClick}>{mockLabel}</TextButton>,
    );

    expect(tree).toMatchSnapshot();
  });
});
