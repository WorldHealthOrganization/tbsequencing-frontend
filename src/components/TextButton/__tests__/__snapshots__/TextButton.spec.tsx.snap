// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextButton component test Should render correctly 1`] = `
<button
  className="css-3xplu6-TextButton"
  disabled={false}
  onClick={[MockFunction]}
  type="button"
>
  <p
    className="MuiTypography-root MuiTypography-body1 css-vax9pj-MuiTypography-root-BaseText"
  >
    mockLabel
  </p>
</button>
`;
